libyul/AsmJsonConverter.cpp

```diff
@@ -51,23 +52,20 @@ Json AsmJsonConverter::operator()(Literal const& _node) const
 	switch (_node.kind)
 	{
 	case LiteralKind::Number:
-		yulAssert(
-			util::isValidDecimal(_node.value.str()) || util::isValidHex(_node.value.str()),
-			"Invalid number literal"
-		);
+		yulAssert(validNumberLiteral(_node), "Invalid number literal");
 		ret["kind"] = "number";
 		break;
 	case LiteralKind::Boolean:
 		ret["kind"] = "bool";
 		break;
 	case LiteralKind::String:
 		ret["kind"] = "string";
-		ret["hexValue"] = util::toHex(util::asBytes(_node.value.str()));
+		ret["hexValue"] = util::toHex(util::asBytes(formatLiteral(_node)));
 		break;
 	}
 	ret["type"] = _node.type.str();
-	if (util::validateUTF8(_node.value.str()))
-		ret["value"] = _node.value.str();
+	if (auto const value = formatLiteral(_node); util::validateUTF8(value))
```

Comment:
I always liked this kind of notation because it is for me easier to digest as it is a bit more compact, but I can also see your perspective. It's definitely not required to use it :)

Reply:
For me the compactness is exactly what is the problem here. The variable declaration takes the front stage and gets in the way of what I actually want to see in the `if`, i.e. the condition :)

Reply:
For me the compactness is exactly what is the problem here. The variable declaration takes the front stage and gets in the way of what I actually want to see in the `if`, i.e. the condition :)

It's the same trade-off as e.g. putting multiple variables in the same declaration to save space (which is something we don't even support in Solidity for this reason).

I do like the scoping aspect of this kind of `if`. It never seemed worth the decreased readability though.